pkg/graph/schema.graphql

```diff
@@ -1767,6 +1770,180 @@ enum SatisfactionLevel {
+  modifiedBy: UUID
+  modifiedByUserAccount: UserAccount
+  modifiedDts: Time
+
+}
+"""
+User Notification Content represents the possible data associated with a User Notification
+"""
+union UserNotificationContent = PlanDiscussion | DiscussionReply
+"""
+UserNotification represents a notification about a specific Activity
+"""
+type UserNotification {
+  id: UUID!
+	activityID: UUID!
+  # activityType: ActivityType #Maybe add this on this level so the FE doesn't need to query it?
+	userID: UUID! #probably not needed, it should be only visible for the user
+  isRead: Boolean!
+  isArchived: Boolean!
+  emailSent: Boolean!
+  activity: Activity! # should we nest this?
+  content: UserNotificationContent!
+
+  createdBy: UUID!
+  createdByUserAccount: UserAccount!
+  createdDts: Time!
+  modifiedBy: UUID
+  modifiedByUserAccount: UserAccount
+  modifiedDts: Time
+
+}
+
+"""
+This is a wrapper for all information for a user 
+"""
+type UserNotifications {
+"""
+This returns the number of unread notifications
+"""
+numUnreadNotifications: Int!
+"""
+This includes all notifications
+"""
+notifications: [UserNotification!]!
+"""
+This renders only the unread notifications
+"""
+unreadNotifications: [UserNotification!]!
+}
+
+"""
+UserNotificationPreferences represents a users preferences about what type and where to receive a notification
+"""
+type UserNotificationPreferences {
+  id: UUID!
+	userID: UUID!
+
+  dailyDigestCompleteEmail: Boolean!
+  dailyDigestCompleteInApp: Boolean!
+
+  addedAsCollaboratorEmail: Boolean!
+  addedAsCollaboratorInApp: Boolean!
+
+  taggedInDiscussionEmail: Boolean!
+  taggedInDiscussionInApp: Boolean!
+
+  taggedInDiscussionReplyEmail: Boolean!
+  taggedInDiscussionReplyInApp: Boolean!
+
+  newDiscussionReplyEmail: Boolean!
+  newDiscussionReplyInApp: Boolean!
+
+  modelPlanSharedEmail: Boolean!
+  modelPlanSharedInApp: Boolean!
+
+  newPlanDiscussionEmail: Boolean! #TODO: EASI-3925 this is not explicitly requested
+  newPlanDiscussionInApp: Boolean!
+
+
+  createdBy: UUID!
+  createdByUserAccount: UserAccount!
+  createdDts: Time!
+  modifiedBy: UUID
+  modifiedByUserAccount: UserAccount
+  modifiedDts: Time
+
+}
+
+"""
+UserNotificationPreferencesChanges represents the ways that a UserNotifications Preferences object can be updated
+"""
+input UserNotificationPreferencesChanges @goModel(model: "map[string]interface{}")  {
+
+  dailyDigestCompleteEmail: Boolean!
+  dailyDigestCompleteInApp: Boolean!
+
+  addedAsCollaboratorEmail: Boolean!
+  addedAsCollaboratorInApp: Boolean!
+
+  taggedInDiscussionEmail: Boolean!
+  taggedInDiscussionInApp: Boolean!
+
+  taggedInDiscussionReplyEmail: Boolean!
+  taggedInDiscussionReplyInApp: Boolean!
+
+  newDiscussionReplyEmail: Boolean!
+  newDiscussionReplyInApp: Boolean!
+
+  modelPlanSharedEmail: Boolean!
+  modelPlanSharedInApp: Boolean!
+
+  newPlanDiscussionEmail: Boolean! #TODO: EASI-3925 this is not explicitly requested
+  newPlanDiscussionInApp: Boolean!

```

Comment:
Do we intend to require that these are set? The typical pattern is allowing these as optional fields

Reply:
Nope! Actually just pushed that change up a little bit ago, it was a mistake from copying from the initial type.